app/renderer/abr-document.js

```diff
@@ -298,19 +297,32 @@ AbrDocument.prototype = {
         var filenameCapture = /([\w][\w\s-]*)/; // Captures are expensive, so use it only if required
         var specialChar = /[^\w\s-]/g;
 
-        if (this.cm.doc.lineCount() < 1) {
+        if (this.getData().trim().length < 1) {
           return "";
         }
 
-        // Find first usable header or line
-        var titleLine = this.toc.find((header) => filename.test(header));
-        if (titleLine) {
-          var data = this.getData();
-          var lines = data.split(os.EOL);
-          titleLine = lines.find((line) => filename.test(line));
+        // Find first usable header
+        var titleHeader = this.toc.find((header) => filename.test(header.content));
```

Comment:
It's not likely, but it's possible for the user to have a header that doesn't have filename suitable characters. For example:

```markdown
# !!!
## This is a suitable header
```
I can't think of any reasonable user that would do something like that, but since it's *allowed* in the editor, I figured I'd be robust against it. What do you think?